tests/mocked_plans/test_base_plan.py

```diff
@@ -32,3 +33,25 @@ async def test_base_plan(_, sample_plans):
     plan = await generate_plan(analysis_results, args)
 
     assert str(plan) == exp_plan
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.nova_compute.get_instance_count", side_effect=[1, 0, 1])
+async def test_base_plan_non_empty(_, sample_plans, caplog):
+    """Testing non-empty hypervisor logs."""
+    args = CLIargs("plan", auto_approve=True)
+    model, __ = sample_plans["base.yaml"]
+
+    with caplog.at_level(logging.INFO):
+        analysis_results = await Analysis.create(model)
+        await generate_plan(analysis_results, args)
+
+    expected_log = (
+        "Skipped (non-empty) hypervisors: [\"Machine(machine_id='1', apps=('nova-compute', "
+        "'ovn-chassis'), az='az-0')\", \"Machine(machine_id='3', apps=('nova-compute', "
+        "'ovn-chassis'), az='az-0')\"]"
+    )
```

Comment:
this is a little cluttered because hypervisors will by definition host a nova-compute app, and because it's not that relevant for an operator if they also host other units. I suggest tweaking the output to only report the nova-compute unit name. So something like
```
Skipped (non-empty) hypervisors: nova-compute/0, nova-compute/1
```